src/api/uploads/__tests__/UploadsReachability.test.js

```diff
@@ -0,0 +1,139 @@
+import axios from 'axios';
+import { withData } from 'leche';
+import UploadsReachability from '../UploadsReachability';
+import { DEFAULT_HOSTNAME_UPLOAD } from '../../../constants';
+
+let uploadsReachability;
+
+describe('api/uploads/UploadsReachability', () => {
+    const hostUrls = ['host0', 'host1', 'host2', 'host3'];
+    const currentTimeMS = 1000;
+    const validResults = {
+        host0: {
+            isReachable: false,
+            expirationTimestampMS: currentTimeMS + 1000,
+        },
+        host1: {
+            isReachable: true,
+            expirationTimestampMS: currentTimeMS + 2000,
+        },
+    };
+    const validAndInvalidResults = {
+        host0: {
+            isReachable: true,
+            expirationTimestampMS: currentTimeMS - 1,
+        },
+        host2: {
+            isReachable: false,
+            expirationTimestampMS: currentTimeMS + 20,
+        },
+
+        host3: {
+            isReachable: false,
+            expirationTimestampMS: currentTimeMS + 30,
+        },
+    };
+
+    beforeEach(() => {
+        uploadsReachability = new UploadsReachability();
+    });
+
+    describe('isReachable()', () => {
+        withData(
```

Comment:
I don't see `leche` used anywhere else in the repo. Could we accomplish the same thing with `test.each`? https://jestjs.io/docs/en/api#testeachtablename-fn-timeout

Reply:
Fixed